pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -77,6 +78,8 @@
     protected AbstractMetadataStore() {
         this.executor = Executors
                 .newSingleThreadScheduledExecutor(new DefaultThreadFactory("metadata-store"));
+        this.executorMonitor = new ExecutorMonitor(this.executor, TimeUnit.MINUTES.toSeconds(5),
```

Comment:
I suggest a config to enable this feature, and default as false.

Reply:
Can you please elaborate why one would want a config knob for this - why would anyone not want to have the monitoring disabled? Once there is a deadlock, the broker will no longer be functional.

Assuming we still want to add a knob, which part are you suggesting we have config for:
1) monitoring the executor.
2) action to take when deadlock is detected (just log and keep chugging along vs restart the broker).

Reply:
@Jason918 PTAL when you get a chance.

Reply:
> why would anyone not want to have the monitoring disabled?

1. Debug purpose, so that we can attach into the process and dig deeper.
2. IMO, currently we want to add this because there are some potential deadlock bug. Not some known issue that can not be fixed. Once we are confident these kind of issue won't happen again, we won't need this.

Reply:
1. If one is able to attach to the process (like a developer), then they should have access to the code and be able to comment out the code. I don't see the knob adding much value to a "user of the Pulsar cluster". The worry I have is the blowing up of plethora of knobs, which over time becomes unmanageable and untestable. This is a small enhancement for which knob does not make sense, IMO. @merlimat @lhotari please chime in if you have an opinion. If others agree, I will go ahead and add the knob.

2. Even if we fix all known deadlock issues, one may still want to have the monitoring enabled to make sure that new code changes don't introduce new deadlock issues.

Reply:
@Jason918 @merlimat @lhotari please comment when you get a chance. Thank you!

Reply:
@bharanic-dev I think you can start a discussion about this change in the mail list so that more people can get involved.

Reply:
Personally, I am agreed with this watchdog but just need a switch in case something goes wrong with this self-killing feature.

Reply:
@Jason918 I added a knob. PTAL when you get a chance.

Reply:
@Jason918 can you please take a look when you get a chance.

Reply:
@bharanic-dev #15155 seems to be a more common solution, PTAL